### PR TITLE
[FLOSS T1] shellrecorder: fix/extends two shell recorder checks

### PR DESCRIPTION
**Pull request by joni1993**
Some shell recorder were failing (only on manual execution see https://github.com/ElektraInitiative/libelektra/issues/4702)

This PR fixes / extends some of them:

- [x] mathcheck.esr
- [x] selftest.esr

Regarding profiletest.esr and listtest.esr - these also fail but i am not sure what they are intended to test.

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: tests/shell/shell_recorder/selftest.esr [hunk 1]
```diff
@@ -32,10 +35,15 @@ STDOUT: Create a new key user:/test/\/evil\\\/tripleEscaped\\/doubleEscaped/key/
 STDOUT: [section]⏎key="multi⏎line"
 < cat `kdb file $Mountpoint`
 
-WARNINGS: 175,175
+WARNINGS: C03100,C03100
```
Comment (0x6178656c) [inline tests/shell/shell_recorder/selftest.esr:38]: Interesting. This seems to still cause a failure:

```text
CMD: kdb ls user:/tests
RET: 0
STDERR:  Sorry, 2 warnings were issued ;(
 1: Module mini issued the warning C03100:
        Validation Syntactic: Line 1: '[section]' is not a valid key value pair
 2: Module mini issued the warning C03100:
        Validation Syntactic: Line 3: 'line"' is not a valid key value pair
STDOUT: user:/tests/key
=== FAILED Warnings do not match expected pattern C03100,C03100
```

But the warning codes seem to match your change. Not sure what is going on.

**Comment by joni1993** [comment]
Check if you are on latest master and have my PR from yesterday included: https://github.com/ElektraInitiative/libelektra/pull/4703

**Comment by 0x6178656c** [comment]
I included cb1d61780f233e2fca4f26623b89c71b3a4ae20c in #4709 to make the tests pass.

**Comment by markus2330** [comment]
Great teamwork happens here! @0x6178656c can I merge this and you rebase #4709?

**Comment by 0x6178656c** [comment]
> Great teamwork happens here! @0x6178656c can I merge this and you rebase #4709?

Works for me.

**Pull request merged**